File: typescript/lesson-3.ts
// Lesson 3: Conditions (if/else)

// define the threshold
const upThreshold: number = 50000
const downThreshold: number = 10000

// function to insert the number and run the if else logic
function checkPaymentLimit(
    amount: number
): string { // need to have the "string" in them to define what it will return
    if (amount > upThreshold) {
        return "requires manual review" // return is needed
    } else if (amount >= downThreshold && amount <= upThreshold
    ) {
        return "auto approved"
    } else { // directly give the value for else conditions
        return "amount too low"
    }
}

console.log(checkPaymentLimit(100000000)) // should say: requires manual review
console.log(checkPaymentLimit(50000))     // should say: auto approved
console.log(checkPaymentLimit(5000))      // should say: amount too low

File: golang/lesson-3.go
package main

import "fmt"

var upLim int = 50000000
var bottomLim int = 10000

func checkPaymentLimit(amount int) string { // put the type of return here
	if amount > upLim {
		return "manual review"
	} else if amount > bottomLim && amount <= upLim { // else if need to be in the same with { bracket
		return "auto approved"
	} else {
		return "amount too low"
	}
}

func main() {
	fmt.Println(checkPaymentLimit(100000000))
	fmt.Println(checkPaymentLimit(2000000))
	fmt.Println(checkPaymentLimit(100))

	paymentMethod := "QRIS" // this can be direct, technically can be paymentMethod string =, but this is faster
	switch paymentMethod {
	case "QRIS":
		fmt.Println("Pay with QRIS")
	case "DANA":
		fmt.Println("Pay with DANA")
	case "OVO":
		fmt.Println("Pay with OVO")
	default:
		fmt.Println("Other Method")
	}
}
